trunk/SUAVE/Components/Energy/Cooling/Cryocooler.py

```diff
@@ -0,0 +1,132 @@
+## @ingroup Components-Energy-Cooling
+# Cryocooler.py
+# 
+# Created:  Feb 2020,   K.Hamilton
+# Modified: Nov 2021,   S. Claridge
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+# suave imports
+import SUAVE
+
+# package imports
+from SUAVE.Core import Data
+from SUAVE.Components.Energy.Energy_Component import Energy_Component
+import numpy as np
+
+# ----------------------------------------------------------------------
+#  Cryocooler
+# ----------------------------------------------------------------------    
+## @ingroup Components-Energy-Cooling-Cryocooler
+class Cryocooler(Energy_Component):
+    
+    """
+    Cryocooler provides cooling power to cryogenic components.
+    Energy is used by this component to provide the cooling, despite the cooling power provided also being an energy inflow.
+    """
+    def __defaults__(self):
+        
+        # Initialise cryocooler properties as null values
+        self.cooler_type        = ''
+        self.rated_power        =   0.0
+        self.min_cryo_temp      =   0.0 
+        self.ambient_temp       = 300.0
+
+    def energy_calc(self, cooling_power, cryo_temp, amb_temp):
+
+        """ Calculate the power required by the cryocooler based on the cryocooler type, the required cooling power, and the temperature conditions.
+    
+    Assumptions:
+        Based on mass data for Cryomech cryocoolers as per the datasheets for ground based non-massreduced coolers available via the cryomech website: https://www.cryomech.com/cryocoolers/.
+        The mass is calculated for the requested power level, the cryocooler should be sized for the maximum power level required as its mass will not change during the flight.
+        The efficiency scales with required cooling power and temperature only.
+        The temperature difference and efficiency are taken not to scale with ambient temperature. This should not matter in the narrow range of temperatures in which aircraft operate, i.e. for ambient temperatures between -50 and 50 C.
+        
+    Source: 
+        https://www.cryomech.com/cryocoolers/
+        
+    Inputs:
+
```

Comment:
Great work, similar to your other scripts, kindly convert this script to table format by aligning [watts] with [kelvin]